nav2_costmap_2d/src/costmap_2d.cpp

```diff
@@ -57,6 +59,37 @@ Costmap2D::Costmap2D(
   resetMaps();
 }
 
+Costmap2D::Costmap2D(const nav_msgs::msg::OccupancyGrid & map)
+: default_value_(FREE_SPACE)
+{
+  access_ = new mutex_t();
+
+  // fill local variables
+  size_x_ = map.info.width;
+  size_y_ = map.info.height;
+  resolution_ = map.info.resolution;
+  origin_x_ = map.info.origin.position.x;
+  origin_y_ = map.info.origin.position.y;
+
+  // create the costmap
+  costmap_ = new unsigned char[size_x_ * size_y_];
+
+  // fill the costmap with a data
+  int8_t data;
+  for (unsigned int it = 0; it < size_x_ * size_y_; it++) {
+    data = map.data[it];
+    if (data == nav2_util::OCC_GRID_UNKNOWN) {
+      costmap_[it] = NO_INFORMATION;
+    } else {
+      // Linear conversion from OccupancyGrid data range [OCC_GRID_FREE..OCC_GRID_OCCUPIED]
+      // to costmap data range [FREE_SPACE..LETHAL_OBSTACLE]
+      costmap_[it] = std::round(
+        1.0 * data * (LETHAL_OBSTACLE - FREE_SPACE) /
```

Comment:
Is that 1.0 necessary? If you're trying to cast into doubles, you should do so explicitly with a cast